### PR TITLE
Turning in GeneFinder Week 2

### PR DESCRIPTION
**Pull request by SeunginLyu**
I really enjoyed this first mini project. Please give me any advice on how to make this code better!

### CODE REVIEW

File: gene_finder.py [hunk 1]
```diff
@@ -1,8 +1,9 @@
 # -*- coding: utf-8 -*-
 """
-YOUR HEADER COMMENT HERE
+The gene_finder function recieves a DNA sequence as input,
+then outputs snippets of DNA that are likely to be protein-coding genes
```
Comment (SungwooPark) [inline gene_finder.py]: Nice use of header comment here to give a brief description of what this file does.

File: gene_finder.py [hunk 2]
```diff
@@ -24,14 +25,40 @@ def get_complement(nucleotide):
...
     'T'
     >>> get_complement('C')
     'G'
+
+    #checks the complements backward
+    >>> get_complement(get_complement('A'))
+    'A'
+    >>> get_complement(get_complement('C'))
+    'C'
     """
```
Comment (SungwooPark) [inline gene_finder.py]: Good test cases here.

File: gene_finder.py [hunk 3]
```diff
@@ -45,8 +72,11 @@ def get_reverse_complement(dna):
     >>> get_reverse_complement("CCGCGTTCA")
     'TGAACGCGG'
     """
-    # TODO: implement this
```
Comment (SungwooPark) [inline gene_finder.py]: Test cases?

File: gene_finder.py [hunk 4]
```diff
@@ -108,19 +207,32 @@ def find_all_ORFs_both_strands(dna):
...
+
+    >>> longest_ORF("ATGCATGAATGTAG")
+    'ATGCATGAATGTAG'
     """
-    # TODO: implement this
-    pass
+    ORFs = find_all_ORFs_both_strands(dna)
+    longest = ''
+    # computes the longest ORF by comparing each ORF with temporary longest ORF
+    for ORF in ORFs:
```
Comment (SungwooPark) [inline gene_finder.py]: Just a stylistic suggestion, a variable name that is all capital letters is reserved for a constant.

**Comment by SungwooPark** [comment]
Looks good overall. Great use of test cases. One thing to note though: your code works fine in python2 but not in python3. I think the issue stems from the fact that you are doing some division with list indices and in python3, division operation defaults to floating point division. Therefore, you are getting floating point indices sometimes and this is causing an error.

Our expectation was for everyone to develop in python3, so next time try to use python3.

**Comment by SeunginLyu** [comment]
Thanks for the comments! I was using 'Script' package for Atom to run my codes. What happened was that it was automatically using python2 to run the codes(which I didn't realize when doing this MP1. This why you should probably at least try to run your code using the terminal commands..). But I figured this out when I was doing MP2. So now I'm coding in python3 :)

**Pull request closed without merging**